src/arrayMethodSort.js

```diff
@@ -4,9 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = stringSort) {
+    let marker = 1;
+
+    while (marker !== 0) {
+      marker = 0;
```

Comment:
Use `do.. while` instead of assigning `marker = 1` and check `while (marker > 0)`